config/routes.rb

```diff
@@ -9,4 +9,8 @@
     resources :posts, controller: "users/posts", only: [:index]
   end
 
+  resources :posts, :comments do
```

Comment:
This definitely works, but is pretty unusual. I would expect to see `resources :posts` and `resources comments` blocks separately.